```shell
python "/share/compling/projects/sanpi/script/sample_pickle.py" \
    -p "${DATA_DIR}neg-bigrams_thr0-001p.35f.pkl.gz" \
    -f 'neg_deprel!=neg' -f 'neg_deprel!=advmod' -f 'neg_deprel!=nsubj' -f 'neg_deprel!=case' \
    -f 'neg_deprel!=cc' -f 'neg_deprel!=cc:preconj' -f 'neg_deprel!=cop' -f 'neg_deprel!=det' \
    -c 'neg_deprel' -c 'neg_form_lower' -c 'adv_form_lower' -c 'adj_form_lower' -c 'text_window' \
    -f 'neg_head==ADJ' -N 10000 -q -C > "${DATA_DIR}sample-35f_unk-deprel_adj-head.10k.csv"
```

In [1]:
import pandas as pd
from pathlib import Path
from source.utils.sample import sample_pickle

DATA_DIR=Path("/share/compling/data/sanpi/4_post-processed/RBdirect")
COLUMNS=['neg_deprel', 'neg_form_lower', 'adv_form_lower', 'adj_form_lower', 'text_window']
HEAD = 'ADJ'
N_FILES_PROC = 35
N_ROWS = 10000

#> Loaded variable 'df' from URI: /share/compling/data/sanpi/4_post-processed/RBdirect/sample-35f_unk-deprel_adj-head.1k.csv
# df = pd.read_csv(r'/share/compling/data/sanpi/4_post-processed/RBdirect/sample-35f_unk-deprel_adj-head.5k.csv')


## Define permitted `neg_deprel` values by head node

From [Oct 23 log](https://gist.github.com/AndreaTheLinguist/ef895a6e8602a9a82ff729f219bc5628#running-notes-monday-october-23-2023):

### `ADJ` _head_

i.e., `NEG` ⬅️ `ADJ`

> | permit ✅                                                         | prohibit ⛔                                              |
> |--------------------------------------------------------------------|----------------------------------------------------------|
> | `neg`, `advmod`, `nsubj`, `case`, `cop`, `det`, `cc`, `cc:preconj` | `advcl`, `amod`, `discourse`, `prep`, `obl`, `obl:npmod` |

In [2]:
OK_DEPREL = {
    'ADJ': ['neg', 'advmod', 'nsubj', 'case', 'cop', 'det', 'cc', 'cc:preconj'], 
    'NEG': ['amod', 'ccomp', 'csubj', 'dep', 'infomod', 'pcomp', 'pobj']
    }

## Load data from composite cleaned and filtered table of hit info

### For 10 corpus parts and only 8 rows first, to illustrate usage of `sample_pickle`

Note that any sorting (as specified by `sort_by` parameter) is applie _after_ sample is drawn

In [3]:
def set_input_path(n_files: int = N_FILES_PROC) -> Path:
    return DATA_DIR.joinpath(f'neg-bigrams_thr0-001p.{n_files}f.pkl.gz')


def set_filters(head: str = HEAD,
                ok_deprel: dict = OK_DEPREL) -> list:

    # filters = [f'neg_deprel!=neg', 'neg_deprel!=advmod', 'neg_deprel!=nsubj', 'neg_deprel!=case',
    #            'neg_deprel!=cop', 'neg_deprel!=det', 'neg_deprel!=cc', 'neg_deprel!=cc:preconj'
    #            ]
    filters = [f'neg_deprel!={rel}' for rel in ok_deprel[head]]
    return filters + [f'neg_head=={head}']


def get_sample(data_path, filters, print_table, sample_size) -> pd.DataFrame:
    return sample_pickle(
        data_path=data_path,
        sample_size=sample_size,
        columns=COLUMNS,
        filters=filters,
        markdown=True,
        sort_by='neg_deprel',
        print_sample=print_table
    )


df = get_sample(data_path=set_input_path(n_files=10),
                filters=set_filters(),
                print_table=True,
                sample_size=8)


## Sampling from `/share/compling/data/sanpi/4_post-processed/RBdirect/neg-bigrams_thr0-001p.10f.pkl.gz`

- *filtering rows...*
  - ✓ Applied filter: `neg_deprel!=neg`
  - ✓ Applied filter: `neg_deprel!=advmod`
  - ✓ Applied filter: `neg_deprel!=nsubj`
  - ✓ Applied filter: `neg_deprel!=case`
  - ✓ Applied filter: `neg_deprel!=cop`
  - ✓ Applied filter: `neg_deprel!=det`
  - ✓ Applied filter: `neg_deprel!=cc`
  - ✓ Applied filter: `neg_deprel!=cc:preconj`
  - ✓ Applied filter: `neg_head==ADJ`

- *selecting columns...*

### 8 random rows matching filter(s) from `neg-bigrams_thr0-001p.10f.pkl.gz`

| hit_id                                    | neg_deprel   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                              |
|:------------------------------------------|:-------------|:-----------------|:-----------------|:-----------------|:-------------------------------------------------------------------------|
| nyt_eng_


## Sampling from `/share/compling/data/sanpi/4_post-processed/RBdirect/neg-bigrams_thr0-001p.10f.pkl.gz`

- *filtering rows...*
  - ✓ Applied filter: `neg_deprel!=neg`
  - ✓ Applied filter: `neg_deprel!=advmod`
  - ✓ Applied filter: `neg_deprel!=nsubj`
  - ✓ Applied filter: `neg_deprel!=case`
  - ✓ Applied filter: `neg_deprel!=cop`
  - ✓ Applied filter: `neg_deprel!=det`
  - ✓ Applied filter: `neg_deprel!=cc`
  - ✓ Applied filter: `neg_deprel!=cc:preconj`
  - ✓ Applied filter: `neg_head==ADJ`

- *selecting columns...*

### 8 random rows matching filter(s) from `neg-bigrams_thr0-001p.10f.pkl.gz`

| hit_id                                   | neg_deprel   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                        |
|:-----------------------------------------|:-------------|:-----------------|:-----------------|:-----------------|:-------------------------------------------------------------------------------------------------------------------|
| apw_eng_20020208_0776_6:24-36-37         | dep          | without          | more             | difficult        | this course because , without them , the new equilibrium that we are seeking will be more difficult to attain , '' |
| nyt_eng_19990416_0358_28:08-11-12        | dep          | neither          | particularly     | complex          | n't easy , but neither was it particularly complex .                                                               |
| apw_eng_20090522_0774_13:2-3-4           | dep          | no               | more             | clear            | with no more clear chances for sprinters ,                                                                         |
| apw_eng_19970501_1204_15:24-25-26        | dep          | no               | longer           | afraid           | the Conservatives _ and no longer afraid of Labor .                                                                |
| pcc_eng_20_099.6848_x1594566_07:29-31-32 | mark         | without          | as               | blatant          | of the scenes ( without being as blatant as traditional porn )                                                     |
| pcc_eng_06_012.5761_x0187070_041:6-8-9   | mark         | without          | too              | sweet            | is rich and complex without being too sweet and comes in a                                                         |
| pcc_eng_19_050.8210_x0804180_82:24-26-27 | mark         | without          | consciously      | aware            | greater quantity , sometimes without being consciously aware that learning is taking                               |
| pcc_eng_18_046.7718_x0740811_07:25-27-28 | mark         | without          | too              | overwhelming     | the students to shore without being too overwhelming .                                                             |


In [4]:
def clean_data(df):
    if 'hit_id' in df.columns:
        hit_id = df.hit_id
        df = df.set_index('hit_id')
    else: 
        hit_id = df.index.to_series()

    df['corpus'] = hit_id.str.split('_eng_').str.get(0)
    df['match_ix'] = hit_id.str.rsplit(':',1).str.get(1)
    return df

df_clean = clean_data(df.copy())
df_clean.head()

neg_deprel neg_form_lower adv_form_lower  \
hit_id                                                                       
nyt_eng_19960321_0496_37:3-7-8           dep            not            too   
apw_eng_20031202_0056_6:20-21-22         dep            not  statistically   
nyt_eng_19940909_0364_43:7-8-9           dep        nothing            too   
nyt_eng_19960925_0129_20:21-22-23        dep             no           more   
nyt_eng_19940906_0236_11:12-13-14        dep            not             so   

                                  adj_form_lower  \
hit_id                                             
nyt_eng_19960321_0496_37:3-7-8             timid   
apw_eng_20031202_0056_6:20-21-22     significant   
nyt_eng_19940909_0364_43:7-8-9           special   
nyt_eng_19960925_0129_20:21-22-23         insane   
nyt_eng_19940906_0236_11:12-13-14           much   

                                                                         text_window  \
hit_id                                                                                 
nyt_eng_19960321_0496_37:3-7-8     they failed not because they were too timid bu...   
apw_eng_20031202_0056_6:20-21-22   point increase _ also not statistically signif...   
nyt_eng_19940909_0364_43:7-8-9               's good -- but nothing too special . ''   
nyt_eng_19960925_0129_20:21-22-23  or , perhaps , no more insane than anyone else...   
nyt_eng_19940906_0236_11:12-13-14  other hand , seems not so much concerned with ...   

                                  corpus  match_ix  
hit_id                                              
nyt_eng_19960321_0496_37:3-7-8       nyt     3-7-8  
apw_eng_20031202_0056_6:20-21-22     apw  20-21-22  
nyt_eng_19940909_0364_43:7-8-9       nyt     7-8-9  
nyt_eng_19960925_0129_20:21-22-23    nyt  21-22-23  
nyt_eng_19940906_0236_11:12-13-14    nyt  12-13-14

### Now with larger sample and printing suppressed

In [5]:
df = get_sample(data_path=set_input_path(n_files=N_FILES_PROC),
                filters=set_filters(),
                print_table=False,
                sample_size=10000)
df_clean = clean_data(df.copy())


## Sampling from `/share/compling/data/sanpi/4_post-processed/RBdirect/neg-bigrams_thr0-001p.35f.pkl.gz`

- *filtering rows...*
  - ✓ Applied filter: `neg_deprel!=neg`
  - ✓ Applied filter: `neg_deprel!=advmod`
  - ✓ Applied filter: `neg_deprel!=nsubj`
  - ✓ Applied filter: `neg_deprel!=case`
  - ✓ Applied filter: `neg_deprel!=cop`
  - ✓ Applied filter: `neg_deprel!=det`
  - ✓ Applied filter: `neg_deprel!=cc`
  - ✓ Applied filter: `neg_deprel!=cc:preconj`
  - ✓ Applied filter: `neg_head==ADJ`

- *selecting columns...*

### 10000 random rows matching filter(s) from `neg-bigrams_thr0-001p.35f.pkl.gz`




## Sampling from `/share/compling/data/sanpi/4_post-processed/RBdirect/neg-bigrams_thr0-001p.35f.pkl.gz`

- *filtering rows...*
  - ✓ Applied filter: `neg_deprel!=neg`
  - ✓ Applied filter: `neg_deprel!=advmod`
  - ✓ Applied filter: `neg_deprel!=nsubj`
  - ✓ Applied filter: `neg_deprel!=case`
  - ✓ Applied filter: `neg_deprel!=cop`
  - ✓ Applied filter: `neg_deprel!=det`
  - ✓ Applied filter: `neg_deprel!=cc`
  - ✓ Applied filter: `neg_deprel!=cc:preconj`
  - ✓ Applied filter: `neg_head==ADJ`

- *selecting columns...*

### 10000 random rows matching filter(s) from `neg-bigrams_thr0-001p.35f.pkl.gz`

In [6]:
df_clean

neg_deprel neg_form_lower  \
hit_id                                                               
nyt_eng_20050106_0331_51:3-4-5                 attr           none   
nyt_eng_19980807_0307_7:6-7-8                  attr        nothing   
nyt_eng_20010601_0300_29:6-7-8                 attr            not   
nyt_eng_20000824_0262_32:06-39-40              attr        nothing   
nyt_eng_20020303_0225_25:03-09-10              attr        nothing   
...                                             ...            ...   
pcc_eng_07_061.3782_x0975972_094:1-5-6     vocative        nothing   
pcc_eng_19_027.9735_x0435427_58:1-5-6      vocative        nothing   
pcc_eng_06_034.7520_x0545912_06:1-5-6      vocative           none   
pcc_eng_25_068.0248_x1085820_280:1-5-6     vocative         nobody   
pcc_eng_14_041.4575_x0653684_36:07-11-12   vocative        nothing   

                                         adv_form_lower adj_form_lower  \
hit_id                                                                   
nyt_eng_20050106_0331_51:3-4-5                      too        pleased   
nyt_eng_19980807_0307_7:6-7-8                    really            new   
nyt_eng_20010601_0300_29:6-7-8                      too           lame   
nyt_eng_20000824_0262_32:06-39-40                    so      authentic   
nyt_eng_20020303_0225_25:03-09-10                   too          major   
...                                                 ...            ...   
pcc_eng_07_061.3782_x0975972_094:1-5-6         probably         asleep   
pcc_eng_19_027.9735_x0435427_58:1-5-6             quite       thankful   
pcc_eng_06_034.7520_x0545912_06:1-5-6               n't        serious   
pcc_eng_25_068.0248_x1085820_280:1-5-6             ever           good   
pcc_eng_14_041.4575_x0653684_36:07-11-12        totally    nonsensical   

                                                                                text_window  \
hit_id                                                                                        
nyt_eng_20050106_0331_51:3-4-5                     critics were none too pleased , either .   
nyt_eng_19980807_0307_7:6-7-8                   it the economy ? Nothing really new there .   
nyt_eng_20010601_0300_29:6-7-8                      lame is it ? Not too lame , but not too   
nyt_eng_20000824_0262_32:06-39-40         day , it was nothing more taxing than swimming...   
nyt_eng_20020303_0225_25:03-09-10         it 's nothing , I 'm thinking , too major righ...   
...                                                                                     ...   
pcc_eng_07_061.3782_x0975972_094:1-5-6      Nothing , she 's probably asleep , and when she   
pcc_eng_19_027.9735_x0435427_58:1-5-6          Nothing , I am quite thankful for the way my   
pcc_eng_06_034.7520_x0545912_06:1-5-6                           None , it was n't serious .   
pcc_eng_25_068.0248_x1085820_280:1-5-6                 Nobody but you was ever good to me .   
pcc_eng_14_041.4575_x0653684_36:07-11-12  's like , ' Nothing , it 's totally nonsensica...   

                                         corpus  match_ix  
hit_id                                                     
nyt_eng_20050106_0331_51:3-4-5              nyt     3-4-5  
nyt_eng_19980807_0307_7:6-7-8               nyt     6-7-8  
nyt_eng_20010601_0300_29:6-7-8              nyt     6-7-8  
nyt_eng_20000824_0262_32:06-39-40           nyt  06-39-40  
nyt_eng_20020303_0225_25:03-09-10           nyt  03-09-10  
...                                         ...       ...  
pcc_eng_07_061.3782_x0975972_094:1-5-6      pcc     1-5-6  
pcc_eng_19_027.9735_x0435427_58:1-5-6       pcc     1-5-6  
pcc_eng_06_034.7520_x0545912_06:1-5-6       pcc     1-5-6  
pcc_eng_25_068.0248_x1085820_280:1-5-6      pcc     1-5-6  
pcc_eng_14_041.4575_x0653684_36:07-11-12    pcc  07-11-12  

[10000 rows x 7 columns]

In [7]:
deprel_counts=df_clean.value_counts('neg_deprel', normalize=True)
print(deprel_counts.multiply(100).round(2).to_frame('% of 10k ADJ-head sample').to_markdown(floatfmt=',.2f'))

| neg_deprel   |   % of 10k ADJ-head sample |
|:-------------|---------------------------:|
| mark         |                      53.58 |
| dep          |                      40.11 |
| conj         |                       3.10 |
| preconj      |                       1.43 |
| parataxis    |                       0.53 |
| aux          |                       0.34 |
| vocative     |                       0.18 |
| dislocated   |                       0.15 |
| rel          |                       0.14 |
| nsubjpass    |                       0.08 |
| attr         |                       0.08 |
| punct        |                       0.06 |
| csubj        |                       0.05 |
| ccomp        |                       0.04 |
| reparandum   |                       0.04 |
| obj          |                       0.03 |
| compound     |                       0.03 |
| expl         |                       0.02 |
| partmod      |                       0.01 |


| neg_deprel   |   % of 10k ADJ-head sample |
|:-------------|---------------------------:|
| mark         |                      52.96 |
| dep          |                      40.49 |
| conj         |                       3.25 |
| preconj      |                       1.61 |
| parataxis    |                       0.52 |
| aux          |                       0.41 |
| dislocated   |                       0.16 |
| rel          |                       0.13 |
| ccomp        |                       0.10 |
| vocative     |                       0.09 |
| nsubjpass    |                       0.07 |
| reparandum   |                       0.06 |
| attr         |                       0.04 |
| obj          |                       0.03 |
| punct        |                       0.03 |
| expl         |                       0.02 |
| dobj         |                       0.02 |
| compound     |                       0.01 |

In [8]:
for g, d in df_clean[['neg_deprel', 'neg_form_lower', 'corpus']].groupby('neg_deprel'):
    vc = d.value_counts(normalize=True).multiply(100).round(2).to_frame(
        '% of `neg_deprel`')
    vc['count']=d.value_counts()
    vc = vc.reset_index().iloc[:, 1:].set_index('neg_form_lower')

    print(f'`neg_deprel == {g}`: _{len(d)} total in 10k sample ({round(len(d)/len(df_clean) * 100, 2)}%)_')
    print(vc.to_markdown(floatfmt=',.1f'))
    print('\n---\n')

`neg_deprel == attr`: _8 total in 10k sample (0.08%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nothing          | nyt      |                62.5 |       5 |
| none             | nyt      |                25.0 |       2 |
| not              | nyt      |                12.5 |       1 |

---

`neg_deprel == aux`: _34 total in 10k sample (0.34%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| ain't            | pcc      |                88.2 |      30 |
| aint             | pcc      |                11.8 |       4 |

---

`neg_deprel == ccomp`: _4 total in 10k sample (0.04%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| not              | nyt      |                50.0 |       2 |
| none             | nyt      |                25.0 |    

`neg_deprel == attr`: _4 total in 10k sample (0.04%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nothing          | nyt      |                75.0 |       3 |
| none             | nyt      |                25.0 |       1 |

---

`neg_deprel == aux`: _41 total in 10k sample (0.41%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| ain't            | pcc      |                90.2 |      37 |
| aint             | pcc      |                 9.8 |       4 |

---

`neg_deprel == ccomp`: _10 total in 10k sample (0.1%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| not              | nyt      |                40.0 |       4 |
| not              | apw      |                20.0 |       2 |
| none             | nyt      |                10.0 |       1 |
| not              | pcc      |                10.0 |       1 |
| nothing          | nyt      |                10.0 |       1 |
| nothing          | pcc      |                10.0 |       1 |

---

`neg_deprel == compound`: _1 total in 10k sample (0.01%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| none             | pcc      |               100.0 |       1 |

---

`neg_deprel == conj`: _325 total in 10k sample (3.25%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| not              | nyt      |                74.8 |     243 |
| not              | apw      |                22.5 |      73 |
| never            | nyt      |                 0.6 |       2 |
| not              | pcc      |                 0.6 |       2 |
| nothing          | nyt      |                 0.6 |       2 |
| few              | pcc      |                 0.3 |       1 |
| none             | nyt      |                 0.3 |       1 |
| nothing          | pcc      |                 0.3 |       1 |

---

`neg_deprel == dep`: _4049 total in 10k sample (40.49%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| no               | nyt      |                40.1 |    1625 |
| not              | nyt      |                34.4 |    1392 |
| no               | apw      |                 7.5 |     303 |
| not              | apw      |                 6.3 |     255 |
| nothing          | nyt      |                 3.6 |     147 |
| none             | nyt      |                 2.0 |      82 |
| neither          | nyt      |                 1.9 |      77 |
| neither          | apw      |                 1.0 |      41 |
| nothing          | apw      |                 0.9 |      35 |
| few              | nyt      |                 0.5 |      19 |
| none             | apw      |                 0.4 |      15 |
| never            | nyt      |                 0.3 |      14 |
| without          | nyt      |                 0.3 |      13 |
| nor              | nyt      |                 0.2 |      10 |
| without          | apw      |                 0.1 |       6 |
| few              | apw      |                 0.1 |       3 |
| rarely           | nyt      |                 0.1 |       3 |
| n't              | nyt      |                 0.1 |       2 |
| hardly           | nyt      |                 0.1 |       2 |
| barely           | apw      |                 0.0 |       1 |
| nobody           | nyt      |                 0.0 |       1 |
| never            | apw      |                 0.0 |       1 |
| neither          | pcc      |                 0.0 |       1 |
| nobody           | apw      |                 0.0 |       1 |

---

`neg_deprel == dislocated`: _16 total in 10k sample (0.16%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nothing          | pcc      |                81.2 |      13 |
| none             | pcc      |                18.8 |       3 |

---

`neg_deprel == dobj`: _2 total in 10k sample (0.02%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| neither          | nyt      |                50.0 |       1 |
| nothing          | apw      |                50.0 |       1 |

---

`neg_deprel == expl`: _2 total in 10k sample (0.02%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nit              | pcc      |               100.0 |       2 |

---

`neg_deprel == mark`: _5296 total in 10k sample (52.96%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| without          | pcc      |                99.4 |    5265 |
| without          | nyt      |                 0.2 |      12 |
| no               | pcc      |                 0.1 |       6 |
| without          | apw      |                 0.1 |       5 |
| w/o              | pcc      |                 0.1 |       4 |
| na               | pcc      |                 0.1 |       3 |
| aint             | pcc      |                 0.0 |       1 |

---

`neg_deprel == nsubjpass`: _7 total in 10k sample (0.07%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| none             | nyt      |                42.9 |       3 |
| nothing          | nyt      |                42.9 |       3 |
| few              | nyt      |                14.3 |       1 |

---

`neg_deprel == obj`: _3 total in 10k sample (0.03%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| neither          | pcc      |                66.7 |       2 |
| no               | pcc      |                33.3 |       1 |

---

`neg_deprel == parataxis`: _52 total in 10k sample (0.52%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| not              | nyt      |                75.0 |      39 |
| not              | apw      |                17.3 |       9 |
| aint             | pcc      |                 1.9 |       1 |
| few              | apw      |                 1.9 |       1 |
| few              | pcc      |                 1.9 |       1 |
| none             | pcc      |                 1.9 |       1 |

---

`neg_deprel == preconj`: _161 total in 10k sample (1.61%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| neither          | nyt      |                89.4 |     144 |
| neither          | apw      |                10.6 |      17 |

---

`neg_deprel == punct`: _3 total in 10k sample (0.03%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| ain't            | pcc      |               100.0 |       3 |

---

`neg_deprel == rel`: _13 total in 10k sample (0.13%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| without          | nyt      |                69.2 |       9 |
| without          | apw      |                30.8 |       4 |

---

`neg_deprel == reparandum`: _6 total in 10k sample (0.06%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| no               | pcc      |                83.3 |       5 |
| not              | pcc      |                16.7 |       1 |

---

`neg_deprel == vocative`: _9 total in 10k sample (0.09%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nothing          | pcc      |                44.4 |       4 |
| none             | pcc      |                33.3 |       3 |
| nobody           | pcc      |                22.2 |       2 |

In [9]:
df_clean

neg_deprel neg_form_lower  \
hit_id                                                               
nyt_eng_20050106_0331_51:3-4-5                 attr           none   
nyt_eng_19980807_0307_7:6-7-8                  attr        nothing   
nyt_eng_20010601_0300_29:6-7-8                 attr            not   
nyt_eng_20000824_0262_32:06-39-40              attr        nothing   
nyt_eng_20020303_0225_25:03-09-10              attr        nothing   
...                                             ...            ...   
pcc_eng_07_061.3782_x0975972_094:1-5-6     vocative        nothing   
pcc_eng_19_027.9735_x0435427_58:1-5-6      vocative        nothing   
pcc_eng_06_034.7520_x0545912_06:1-5-6      vocative           none   
pcc_eng_25_068.0248_x1085820_280:1-5-6     vocative         nobody   
pcc_eng_14_041.4575_x0653684_36:07-11-12   vocative        nothing   

                                         adv_form_lower adj_form_lower  \
hit_id                                                                   
nyt_eng_20050106_0331_51:3-4-5                      too        pleased   
nyt_eng_19980807_0307_7:6-7-8                    really            new   
nyt_eng_20010601_0300_29:6-7-8                      too           lame   
nyt_eng_20000824_0262_32:06-39-40                    so      authentic   
nyt_eng_20020303_0225_25:03-09-10                   too          major   
...                                                 ...            ...   
pcc_eng_07_061.3782_x0975972_094:1-5-6         probably         asleep   
pcc_eng_19_027.9735_x0435427_58:1-5-6             quite       thankful   
pcc_eng_06_034.7520_x0545912_06:1-5-6               n't        serious   
pcc_eng_25_068.0248_x1085820_280:1-5-6             ever           good   
pcc_eng_14_041.4575_x0653684_36:07-11-12        totally    nonsensical   

                                                                                text_window  \
hit_id                                                                                        
nyt_eng_20050106_0331_51:3-4-5                     critics were none too pleased , either .   
nyt_eng_19980807_0307_7:6-7-8                   it the economy ? Nothing really new there .   
nyt_eng_20010601_0300_29:6-7-8                      lame is it ? Not too lame , but not too   
nyt_eng_20000824_0262_32:06-39-40         day , it was nothing more taxing than swimming...   
nyt_eng_20020303_0225_25:03-09-10         it 's nothing , I 'm thinking , too major righ...   
...                                                                                     ...   
pcc_eng_07_061.3782_x0975972_094:1-5-6      Nothing , she 's probably asleep , and when she   
pcc_eng_19_027.9735_x0435427_58:1-5-6          Nothing , I am quite thankful for the way my   
pcc_eng_06_034.7520_x0545912_06:1-5-6                           None , it was n't serious .   
pcc_eng_25_068.0248_x1085820_280:1-5-6                 Nobody but you was ever good to me .   
pcc_eng_14_041.4575_x0653684_36:07-11-12  's like , ' Nothing , it 's totally nonsensica...   

                                         corpus  match_ix  
hit_id                                                     
nyt_eng_20050106_0331_51:3-4-5              nyt     3-4-5  
nyt_eng_19980807_0307_7:6-7-8               nyt     6-7-8  
nyt_eng_20010601_0300_29:6-7-8              nyt     6-7-8  
nyt_eng_20000824_0262_32:06-39-40           nyt  06-39-40  
nyt_eng_20020303_0225_25:03-09-10           nyt  03-09-10  
...                                         ...       ...  
pcc_eng_07_061.3782_x0975972_094:1-5-6      pcc     1-5-6  
pcc_eng_19_027.9735_x0435427_58:1-5-6       pcc     1-5-6  
pcc_eng_06_034.7520_x0545912_06:1-5-6       pcc     1-5-6  
pcc_eng_25_068.0248_x1085820_280:1-5-6      pcc     1-5-6  
pcc_eng_14_041.4575_x0653684_36:07-11-12    pcc  07-11-12  

[10000 rows x 7 columns]

In [14]:
for deprel in deprel_counts.index:
    
    # print(f'\n- **P** `{deprel}`')
    print(f'\n#### Sample of `{deprel}` relations (with `ADJ` head)\n')
    selection = df_clean[df_clean.neg_deprel==deprel]
    print(selection.sample(min(len(selection),25)).filter(regex=r'(lower|window|ix)').set_index('match_ix').sort_values(['neg_form_lower', 'adv_form_lower']).to_markdown())


#### Sample of `mark` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                          |
|:-----------|:-----------------|:-----------------|:-----------------|:-----------------------------------------------------------------------------------------------------|
| 13-15-16   | without          | actually         | friendly         | be " friends " without being actually friendly , a key temptation                                    |
| 15-17-18   | without          | downright        | hostile          | as uncomfortable as possible without being downright hostile .                                       |
| 28-30-31   | without          | explicitly       | negative         | that they were upset without being explicitly negative .                                             |
| 14-16-17   | without          | genuinely        | concerned        | w


#### Sample of `mark` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                          |
|:-----------|:-----------------|:-----------------|:-----------------|:-----------------------------------------------------------------------------------------------------|
| 13-15-16   | without          | actually         | friendly         | be " friends " without being actually friendly , a key temptation                                    |
| 15-17-18   | without          | downright        | hostile          | as uncomfortable as possible without being downright hostile .                                       |
| 28-30-31   | without          | explicitly       | negative         | that they were upset without being explicitly negative .                                             |
| 14-16-17   | without          | genuinely        | concerned        | we cannot do this without being genuinely concerned for our neighbors most                           |
| 22-29-30   | without          | less             | depressed        | more depressed than women without while men with job authority are less depressed than men without . |
| 19-21-22   | without          | more             | specific         | Taketoshi Tanoue said , without being more specific about the offenses .                             |
| 28-31-32   | without          | more             | beneficial       | within rather than from without may be more beneficial in the end .                                  |
| 16-19-20   | without          | n't              | worth            | part of my world without it is n't worth living , " Hotaru                                           |
| 14-16-17   | without          | overly           | forceful         | the here and now without being overly forceful in its energy before                                  |
| 21-27-28   | without          | slightly         | careful          | lost in someone else without being , at least , slightly careful .                                   |
| 12-17-18   | without          | so               | good             | a melancholy prospect ; without your Lordships will be so good as to interpose in                    |
| 12-14-15   | without          | too              | theatrical       | historically accurate and colorful without being too theatrical .                                    |
| 07-09-10   | without          | too              | hard             | had a nice crunch without being too hard and was dry enough                                          |
| 13-15-16   | without          | too              | chatty           | with - always cheerful without being too chatty , invariably on hand                                 |
| 08-10-11   | without          | too              | fussy            | clean and elegant , without being too fussy and will dress down                                      |
| 18-20-21   | without          | too              | strenuous        | to the higher altitude without being too strenuous .                                                 |
| 15-17-18   | without          | too              | boring           | clear and calm tone without being too boring or stiff .                                              |
| 16-18-19   | without          | too              | loose            | well around the foot without being too loose and learned how many                                    |
| 16-18-19   | without          | too              | overpowering     | pleasing scent to rooms without being too overpowering .                                             |
| 4-6-7      | without          | too              | sweet            | It was sweet without being too sweet and chocolately without being                                   |
| 4-6-7      | without          | too              | muscular         | They are defined without being too muscular .                                                        |
| 07-09-10   | without          | unnecessarily    | pessimistic      | of the slip -- without being unnecessarily pessimistic -- is the looming                             |
| 13-20-21   | without          | utterly          | different        | of Brisbane , and without that setting the tone would be utterly different .                         |
| 25-27-28   | without          | visually         | offensive        | it 's just funny without being visually offensive .                                                  |
| 1-3-4      | without          | wholly           | present          | Without being wholly present , touring a home                                                        |

#### Sample of `dep` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                            |
|:-----------|:-----------------|:-----------------|:-----------------|:---------------------------------------------------------------------------------------|
| 21-22-23   | neither          | sufficiently     | reliable         | `` the information is neither sufficiently reliable not sufficiently specific .        |
| 2-6-7      | no               | just             | happy            | `` No , I 'm just happy that we even reached                                           |
| 7-8-9      | no               | less             | better           | , our ability is no less better than other people 's                                   |
| 09-10-11   | no               | less             | clear            | , the written opinion no less clear :                                                  |
| 3-4-5      | no               | less             | key              | Music is no less key , but far less                                                    |
| 7-8-9      | no               | less             | threatening      | respect , it is no less threatening to religious freedom than                          |
| 19-20-21   | no               | more             | sure             | voters impressed , if no more sure about which ticket would                            |
| 09-10-11   | no               | more             | frequent         | canine looniness is probably no more frequent under a full moon                        |
| 49-50-51   | no               | more             | important        | the first quarter is no more important than fundraising throughout the                 |
| 7-8-9      | no               | more             | yellow           | on the porch were no more yellow than they were on                                     |
| 08-09-10   | no               | more             | ambitious        | more complicated , though no more ambitious , than the Friendship                      |
| 13-14-15   | no               | more             | charismatic      | said he was `` no more charismatic than anybody else ''                                |
| 7-8-9      | no               | more             | legitimate       | draw with Chavez was no more legitimate .                                              |
| 18-22-23   | no               | too              | good             | Me not good ? No , he is too good . ''                                                 |
| 24-25-26   | none             | as               | good             | passage before , but none as good as in the nature                                     |
| 11-12-13   | none             | more             | critical         | endless needs , but none more critical than stable leadership .                        |
| 5-6-7      | not              | always           | available        | `` The food , not always available because of the indebtedness                         |
| 05-10-11   | not              | most             | original         | for the record -- not that this is the most original of observations these days        |
| 08-09-10   | not              | so               | much             | is about desire -- not so much for sex as for                                          |
| 17-22-23   | not              | terribly         | difficult        | school -- a challenge not because the work is terribly difficult , but because she     |
| 05-10-11   | not              | thoroughly       | impressive       | and O'Neal , if not typical , was still thoroughly impressive .                        |
| 13-14-15   | not              | too              | many             | surface defects -- but not too many -- to improve bonding                              |
| 15-20-21   | not              | too              | icky             | for the Farrellys , not because their sensibility is too icky but because their hearts |
| 2-3-4      | not              | yet              | conclusive       | though not yet conclusive , a stream of                                                |
| 07-11-12   | without          | most             | beautiful        | Antarctica is simply , without question , the most beautiful                           |

#### Sample of `conj` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                    |
|:-----------|:-----------------|:-----------------|:-----------------|:---------------------------------------------------------------------------------------------------------------|
| 15-19-20   | not              | any              | good             | events '' whether or not the show is any good . -RRB-                                                          |
| 38-39-40   | not              | as               | capable          | as my mother , not as capable and independent as Maryellen                                                     |
| 15-17-18   | not              | as               | productive       | haphazard at best and not nearly as productive as stream fishing ,                                             |
| 17-18-19   | not              | as               | small            | hero can be , not as small and snarky as he                                                                    |
| 7-8-9      | not              | as               | intense          | was bigger , but not as intense as Charley , ''                                                                |
| 10-11-12   | not              | as               | powerful         | is well-intentioned , but not as powerful or disturbing or inspiring                                           |
| 23-25-26   | not              | as               | manic            | bland and sweet _ not nearly as manic or twisted as they                                                       |
| 15-16-17   | not              | especially       | elusive          | against the run but not especially elusive or overpowering when rushing                                        |
| 16-17-18   | not              | especially       | imaginative      | seems tagged on , not especially imaginative , and smotheringly conventional                                   |
| 12-13-14   | not              | even             | old              | of high school and not even old enough to legally drink                                                        |
| 23-24-25   | not              | exactly          | impressive       | than Worcester , but not exactly impressive .                                                                  |
| 04-09-10   | not              | fully            | safe             | `` Whether or not we can be ever fully safe is up -- you                                                       |
| 08-13-14   | not              | hiv              | positive         | , is whether or not the child is definitely HIV positive or had only tested                                    |
| 29-38-39   | not              | overly           | sensitive        | as `` whether or not Colgate students , majority or minority , are overly sensitive concerning racial issues . |
| 17-18-19   | not              | overtly          | defiant          | demeanor as sullen , not overtly defiant but sarcastic .                                                       |
| 11-17-18   | not              | politically      | motivated        | is irrelevant whether or not President Fox 's actions are politically motivated , '' Chabat said               |
| 08-12-13   | not              | politically      | motivated        | comment on whether or not the protests were politically motivated .                                            |
| 23-24-25   | not              | really           | funny            | just crass commercial and not really funny and well-written . ''                                               |
| 35-36-37   | not              | so               | much             | art seriously , but not so much herself .                                                                      |
| 7-8-9      | not              | so               | hot              | get hotter , but not so hot that they boil away                                                                |
| 12-13-14   | not              | so               | much             | a distraction originally but not so much now , '' said                                                         |
| 13-18-19   | not              | too              | long             | acting and whether or not the three-hour-plus movie is too long .                                              |
| 15-17-18   | not              | too              | young            | a wealthy dilettante and not `` too young , too inexperienced and                                              |
| 17-18-19   | not              | very             | interesting      | very white bread and not very interesting .                                                                    |
| 06-09-10   | without          | even             | much             | understaffed , undersupplied and without air-conditioning or even much lighting , the domed                    |

#### Sample of `preconj` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                       |
|:-----------|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------------------|
| 09-12-13   | neither          | always           | easy             | judge , Alito was neither predictable nor always easy to categorize politically . |
| 27-28-29   | neither          | as               | quick            | Conference -- he 's neither as quick as Stockton nor able                         |
| 15-16-17   | neither          | as               | hilarious        | , Scholastic -RRB- is neither as hilarious nor as cynical ,                       |
| 25-26-27   | neither          | as               | upbeat           | said that he remained neither as upbeat as his advisers like                      |
| 08-09-10   | neither          | as               | pleasant         | , it will be neither as pleasant nor as efficient as                              |
| 21-22-23   | neither          | as               | groovy           | about ? We are neither as groovy nor as British as                                |
| 08-09-10   | neither          | as               | silly            | song about candy was neither as silly nor as little as                            |
| 09-10-11   | neither          | as               | big              | Malhotra , who is neither as big as Laraque nor as                                |
| 09-10-11   | neither          | as               | special          | and our media are neither as special , nor as cursed                              |
| 09-10-11   | neither          | more             | optimistic       | Madeline Albright is `` neither more optimistic nor more pessimistic than         |
| 08-09-10   | neither          | overwhelmingly   | wide             | -- seeming to be neither overwhelmingly wide nor perilously narrow --             |
| 08-09-10   | neither          | so               | frightening      | Lap Kok will be neither so frightening nor so exhilarating .                      |
| 15-16-17   | neither          | suitably         | christian        | implication that he is neither suitably Christian nor American in his             |
| 24-25-26   | neither          | too              | strong           | that the economy is neither too strong nor too weak .                             |
| 3-4-5      | neither          | too              | stupid           | he 's neither too stupid nor very smart ,                                         |
| 08-09-10   | neither          | too              | strong           | Goldilocks economy , '' neither too strong nor too weak ,                         |
| 32-33-34   | neither          | too              | oily             | dense crumb that is neither too oily nor too dry ;                                |
| 4-5-6      | neither          | too              | cold             | `` Inflation is neither too cold , nor too hot                                    |
| 12-13-14   | neither          | too              | thin             | top so it was neither too thin nor too thick ,                                    |
| 4-5-6      | neither          | too              | tired            | but McGrath is neither too tired nor too shocked to                               |
| 19-20-21   | neither          | too              | hot              | '' economy that is neither too hot nor too cold .                                 |
| 24-25-26   | neither          | too              | hot              | describe one that is neither too hot nor too cold .                               |
| 16-17-18   | neither          | too              | dressy           | that work , are neither too dressy nor too plain ,                                |
| 6-7-8      | neither          | too              | light            | 's wry retort seemed neither too light nor too mean ,                             |
| 30-31-32   | neither          | too              | soft             | rice foundation that 's neither too soft nor too compressed .                     |

#### Sample of `parataxis` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                                     |
|:-----------|:-----------------|:-----------------|:-----------------|:--------------------------------------------------------------------------------------------------------------------------------|
| 44-45-46   | ain't            | even             | worried          | much time , He ain't even worried yet , the Hog                                                                                 |
| 2-6-7      | none             | not              | sure             | " none " I 'm not sure what you mean by                                                                                         |
| 05-16-17   | not              | again            | worrisome        | `` Today , if not yet at the crisis stage , fuel costs are once again worrisome . ''                                            |
| 04-18-19   | not              | as               | nutritious       | or , though not as crunchy or tasty , frozen vegetables -LRB- without creamy sauces -RRB- are as nutritious as fresh ones .     |
| 14-24-25   | not              | as               | big              | , which , if not as big as all outdoors , is at least as big as your backyard .                                                 |
| 06-14-15   | not              | as               | vulnerable       | Court reporters , although not as prominent as judges , are just as vulnerable , as we saw                                      |
| 23-27-28   | not              | deeply           | troubling        | connections that , if not illegal , is deeply troubling , '' said Shaun                                                         |
| 06-11-12   | not              | extremely        | rare             | protective mutation , while not common , is not extremely rare .                                                                |
| 08-12-13   | not              | hardly           | unprecedented    | by rowboat , though not routine , is hardly unprecedented .                                                                     |
| 16-26-27   | not              | increasingly     | clear            | current economic times , not as big as they once did -- it becomes increasingly clear that the creative juices                  |
| 13-18-19   | not              | less             | gloomy           | results that , while not strong overall , were less gloomy than expected .                                                      |
| 08-13-14   | not              | more             | upsetting        | can often , although not always , be far more upsetting than being doomed to                                                    |
| 37-49-50   | not              | more             | willing          | acquaintances who , though not necessarily beholden to him in a legal sense , were certainly more willing to accede to his      |
| 37-40-41   | not              | much             | less             | he remained , if not obscure , much less than famous .                                                                          |
| 07-22-23   | not              | nonetheless      | unusual          | career path , though not as spectacular as the president 's rise to power from rural poverty , is nonetheless unusual .         |
| 20-31-32   | not              | nonetheless      | real             | -RRB- that , while not as apparent as they are on most teams , are nonetheless real .                                           |
| 04-10-11   | not              | now              | guilty           | many , if not most , Atlanta drivers are now guilty of the same thing                                                           |
| 06-12-13   | not              | plenty           | powerful         | all-aluminum engines , while not performance power plants , are plenty powerful and wonderfully quiet .                         |
| 08-16-17   | not              | relatively       | unusual          | as this , though not uncommon in the United States , are relatively unusual in France .                                         |
| 10-16-17   | not              | relatively       | easy             | by car , while not exactly a breeze , is relatively easy .                                                                      |
| 04-10-11   | not              | simply           | worth            | many , if not most , U.S. companies were simply worth less after the attack                                                     |
| 12-22-23   | not              | very             | cautious         | which , `` while not in conflict with the available scientific information , were very cautious and exposed the decision-makers |
| 25-30-31   | not              | well             | able             | sitters who , though not currently famous , are well able to pique our curiosity                                                |
| 19-23-24   | nothing          | certainly        | spry             | , who , if nothing else , is certainly spry enough to avoid injury                                                              |
| 04-10-11   | nothing          | more             | attractive       | and , if nothing else , LA sure became more attractive to two sitting Bulls                                                     |

#### Sample of `aux` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                           |
|:-----------|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------|
| 6-7-8      | ain't            | even             | interested       | you sleep , I ain't even interested Couple dead presidents in         |
| 7-8-9      | ain't            | even             | close            | day in the sun ain't even close                                       |
| 17-18-19   | ain't            | even             | mad              | ) SO ... I AIN'T EVEN MAD .                                           |
| 11-12-13   | ain't            | even             | black            | 's addiction those who ain't even black use it .                      |
| 20-21-22   | ain't            | ever             | satisfied        | released ELVIS PRESLEY I ain't ever satisfied JOHN MELLENCAMP ( happy |
| 28-29-30   | ain't            | ever             | satisfied        | " and " I Ain't Ever Satisfied " became boisterous singalongs         |
| 2-3-4      | ain't            | never            | scared           | I Ain't Never Scared ...                                              |
| 81-82-83   | ain't            | never            | scared           | , watch us I ain't never scared to ride though Stay                   |
| 7-8-9      | ain't            | never            | satisfied        | what you do She ain't never satisfied Ai n't that just                |
| 10-11-12   | ain't            | quite            | sure             | of Lucknow and I ain't quite sure .                                   |
| 11-12-13   | ain't            | real             | good             | pony on offense you ain't real good .                                 |
| 12-13-14   | ain't            | really           | alive            | ' , then you ain't really alive .                                     |
| 20-21-22   | ain't            | really           | concerned        | do n't talk I ain't really concerned                                  |
| 36-37-38   | ain't            | really           | scared           | comfortable , now you ain't really scared                             |
| 3-4-5      | ain't            | really           | good             | But you ain't really good                                             |
| 41-42-43   | ain't            | too              | old              | old , said you ain't too old Ai n't too old                           |
| 16-17-18   | ain't            | too              | wrong            | good grades , you ain't too wrong ! )                                 |
| 16-17-18   | ain't            | too              | smart            | City cops , they ain't too smart " .                                  |
| 25-26-27   | ain't            | too              | bright           | steel executives , you ain't too bright .                             |
| 08-09-10   | ain't            | too              | worried          | personal opinion which I ain't too worried about .                    |
| 7-8-9      | ain't            | too              | stupid           | ai n't saying you ain't too stupid ! " someone hollered               |
| 08-09-10   | aint             | even             | close            | love boy , this aint even close                                       |
| 27-28-29   | aint             | even             | sure             | say it , i aint even sure                                             |
| 1-2-3      | aint             | even             | worth            | Aint even worth the bother to list                                    |
| 2-5-6      | aint             | real             | close            | I aint never been real close to nobody , either                       |

#### Sample of `vocative` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                          |
|:-----------|:-----------------|:-----------------|:-----------------|:---------------------------------------------------------------------|
| 1-5-6      | nobody           | all              | busy             | Nobody - they 're all busy at the gym ,                              |
| 1-8-9      | nobody           | damn             | sure             | Nobody I know , that 's for damn sure .                              |
| 1-5-6      | nobody           | ever             | good             | Nobody but you was ever good to me .                                 |
| 1-5-6      | nobody           | ever             | good             | Nobody but you was ever good to me .                                 |
| 2-8-9      | nobody           | more             | shocked          | And nobody , nobody could have been more shocked than Ms. Kinsella . |
| 1-5-6      | nobody           | not              | concerned        | Nobody , I am not concerned at where a person                        |
| 1-5-6      | none             | n't              | serious          | None , it was n't serious .                                          |
| 1-6-7      | none             | very             | straight         | None , the interview was very straight forward .                     |
| 1-5-6      | nothing          | absolutely       | correct          | Nothing , you are absolutely correct .                               |
| 1-5-6      | nothing          | just             | reckless         | Nothing , it 's just reckless .                                      |
| 1-6-7      | nothing          | more             | important        | Nothing , no victory is more important than a human life             |
| 1-5-6      | nothing          | more             | precious         | Nothing , NOTHING is more precious than life .                       |
| 1-5-6      | nothing          | probably         | asleep           | Nothing , she 's probably asleep , and when she                      |
| 1-5-6      | nothing          | quite            | thankful         | Nothing , I am quite thankful for the way my                         |
| 1-5-6      | nothing          | too              | busy             | Nothing , I 'm too busy laughing at the memory                       |
| 1-5-6      | nothing          | too              | busy             | Nothing , he is too busy misinterpreting theological texts from      |
| 07-11-12   | nothing          | totally          | nonsensical      | 's like , ' Nothing , it 's totally nonsensical . "                  |
| 1-5-6      | nothing          | utterly          | brilliant        | Nothing , it 's utterly brilliant at what it does                    |

#### Sample of `dislocated` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                                       |
|:-----------|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------------------------------------------------------------------|
| 03-09-10   | none             | all              | original         | No , none of that , it 's all original .                                                                                          |
| 3-8-9      | none             | n't              | capable          | Why , none ; the screenwriter is n't capable of coming up with                                                                    |
| 2-7-8      | nothing          | extremely        | mad              | " Nothing much , he was extremely mad though ; I 'm                                                                               |
| 2-6-7      | nothing          | just             | boring           | " Nothing , they 're just boring .                                                                                                |
| 05-11-12   | nothing          | mostly           | free             | Besides the rig , nothing , the miner programs are mostly free .                                                                  |
| 01-18-19   | nothing          | much             | better           | Nothing cooked in a wood fired oven tastes like food from a conventional oven , it 's much better .                               |
| 01-12-13   | nothing          | n't              | worth            | Nothing short of complete dedication , otherwise , the costumes are n't worth the work ( in                                       |
| 08-12-13   | nothing          | not              | qualified        | doc replies , " nothing , you 're not qualified . "                                                                               |
| 3-7-8      | nothing          | not              | big              | Most likely nothing , I 'm not big on throwing myself a                                                                           |
| 1-8-9      | nothing          | often            | entertaining     | Nothing against movie tie-ins , they are often entertaining and well written ,                                                    |
| 01-16-17   | nothing          | really           | drunk            | Nothing , the boat is not lying on its side , you 're just all really drunk .                                                     |
| 2-7-8      | nothing          | so               | happy            | " Nothing , I 'm just so happy . "                                                                                                |
| 04-11-12   | nothing          | sublimely        | happy            | " Oh , nothing dear , I 'm just so sublimely happy , that 's all                                                                  |
| 01-21-22   | nothing          | totally          | convinced        | Nothing against that , and I 'm not denying the witness of that passage , however , I 'm not totally convinced of the veracity of |
| 01-10-11   | nothing          | very             | interested       | Nothing against the movies , I was just never very interested in plots about superheroes                                          |

#### Sample of `rel` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                                                   |
|:-----------|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------------------------------------------------------------------------------|
| 21-26-27   | without          | almost           | certain          | its sales tax , without which Orange County is almost certain to default on more                                                              |
| 17-23-24   | without          | almost           | impossible       | of clean-energy technologies , without which major reductions would be almost impossible to achieve .                                         |
| 06-11-12   | without          | as               | bleak            | , the players , without whom March would be as bleak as Afghanistan , have                                                                    |
| 27-40-41   | without          | as               | rich             | , hokey dignity - without which the lifestyles of the rich and famous just would n't be as rich or as famous .                                |
| 11-18-19   | without          | as               | barren           | celebrate the trees , without whose benefits the Earth would be as barren and lifeless as Mars                                                |
| 09-16-17   | without          | even             | higher           | out significant structural adjustments without which the deficit would 've been even higher , '' said Almunia                                 |
| 22-27-28   | without          | extremely        | vulnerable       | Khmer Rouge guerrillas , without whom they would be extremely vulnerable militarily .                                                         |
| 27-31-32   | without          | highly           | unlikely         | Iraq 's neighbors , without whom stability is highly unlikely .                                                                               |
| 10-21-22   | without          | less             | valuable         | activity , but one without which the Internet and the World Wide Web are far less valuable . ''                                               |
| 27-33-34   | without          | little           | more             | with that toasty flavor without which a sparkling wine is little more than effervescent grape juice                                           |
| 14-19-20   | without          | more             | difficult        | of the world , without which it becomes ever more difficult to execute foreign policy                                                         |
| 09-14-15   | without          | more             | mundane          | a subtle heat , without which it is simply more mundane .                                                                                     |
| 16-29-30   | without          | much             | harder           | the Serb military , without whose cooperation the complicated process of maintaining the peace would have been much harder .                  |
| 33-49-50   | without          | vastly           | poorer           | expect from Heyday , without whose journal `` News From Native California '' our knowledge of pre-contact California would be vastly poorer . |

#### Sample of `nsubjpass` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                     |
|:-----------|:-----------------|:-----------------|:-----------------|:------------------------------------------------------------------------------------------------|
| 17-19-20   | few              | as               | influential      | a writer , but few are as influential or well traveled as                                       |
| 2-4-5      | neither          | so               | young            | but neither was so young , so vibrant ,                                                         |
| 2-5-6      | nobody           | as               | consistent       | `` Nobody 's been as consistent as I have over                                                  |
| 12-17-18   | none             | as               | committed        | head coach was that none of the aspirants were as committed to or experienced in                |
| 14-16-17   | none             | as               | good             | said , `` but none are as good or as beautifully served                                         |
| 24-26-27   | none             | more             | heartfelt        | legislative session , and none is more heartfelt and laced with guilt                           |
| 29-37-38   | nothing          | as               | adventurous      | than , musically , nothing R&B in the past decade has been as adventurous , as accomplished and |
| 06-09-10   | nothing          | quite            | right            | it , folks , nothing 's been quite right for the Bruins and                                     |

#### Sample of `attr` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                                                                                                                                |
|:-----------|:-----------------|:-----------------|:-----------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 3-4-5      | none             | too              | pleased          | critics were none too pleased , either .                                                                                                                                                                                   |
| 25-26-27   | none             | too              | pleased          | York magazine executives were none too pleased , either .                                                                                                                                                                  |
| 6-7-8      | not              | too              | lame             | lame is it ? Not too lame , but not too                                                                                                                                                                                    |
| 48-51-52   | nothing          | certainly        | relative         | would prove to be nothing special , certainly relative to their recent glory                                                                                                                                               |
| 6-7-8      | nothing          | really           | new              | it the economy ? Nothing really new there .                                                                                                                                                                                |
| 06-39-40   | nothing          | so               | authentic        | day , it was nothing more taxing than swimming and sunning on the wide white-sand beach , eating fish tacos at a picnic table that overlooked the ocean , then saving up for the Lahaina luau , so authentic in its food , |
| 04-10-11   | nothing          | somewhat         | akin             | the return was nothing but a beat down _ somewhat akin to the ones the                                                                                                                                                     |
| 03-09-10   | nothing          | too              | major            | it 's nothing , I 'm thinking , too major right now . ''                                                                                                                                                                   |

#### Sample of `punct` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                              |
|:-----------|:-----------------|:-----------------|:-----------------|:---------------------------------------------------------|
| 10-12-13   | ain't            | so               | nice             | to mind for me ain't nearly so nice . "                  |
| 16-17-18   | ain't            | so               | bad              | 14:28:45 +0000 reading Bilingualism ain't so Bad ! - ] ] |
| 44-45-46   | ain't            | so               | common           | found that common sense ain't so common . "              |
| 34-35-36   | ain't            | so               | smart            | I found out it ain't so smart , " Grassley says          |
| 33-35-36   | ain't            | that             | great            | of a good thing ain't always that great .                |
| 15-16-17   | ain't            | too              | bad              | Donuts at his disposal ain't too bad .                   |

#### Sample of `csubj` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                       |
|:-----------|:-----------------|:-----------------|:-----------------|:--------------------------------------------------------------------------------------------------|
| 03-14-15   | ain't            | damn             | short            | But it ain't up to me to make you change Life is too damn short to live unhappily                 |
| 04-15-16   | ain't            | particularly     | unusual          | Besides , it ain't like death by infection during the Russian Civil War is particularly unusual : |
| 4-7-8      | ain't            | too              | happy            | The boss man ain't gonna be too happy about that .                                                |
| 08-12-13   | few              | not              | important        | that that they are few in number is not important .                                               |
| 31-39-40   | nothing          | truly            | upsetting        | , but to be nothing more than a minor Indian province was truly upsetting .                       |

#### Sample of `ccomp` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                               |
|:-----------|:-----------------|:-----------------|:-----------------|:------------------------------------------------------------------------------------------|
| 30-36-37   | none             | almost           | attractive       | important , it 's none of our business _ seem almost attractive , even logical .          |
| 2-8-9      | not              | relatively       | cheap            | why not ? `` Dateline '' was relatively cheap , easily scalable and                       |
| 02-12-13   | not              | so               | desperate        | why not ? People with Alzheimer 's and their families are so desperate that they will buy |
| 2-7-8      | not              | so               | scared           | why not ? `` I was so scared , '' Chen said                                               |

#### Sample of `reparandum` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                           |
|:-----------|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------|
| 37-40-41   | no               | never            | gonna            | n't stop loving you No I 'm never gonna let you get away              |
| 07-14-15   | no               | not              | bad              | , the answer is no , common Easter Egg dye is not bad ...             |
| 24-29-30   | no               | not              | sure             | yes , 46 % no , and 12 % not sure .                                   |
| 13-17-18   | nothing          | definitely       | worth            | and , well , nothing , it 's definitely worth the minuscule amount of |

#### Sample of `obj` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                          |
|:-----------|:-----------------|:-----------------|:-----------------|:-----------------------------------------------------|
| 26-27-28   | few              | well             | worth            | that there are a few well worth your time .          |
| 37-40-41   | no               | not              | afraid           | That prophecy says Am No I 'm not afraid FG No I 'm  |
| 11-14-15   | no               | not              | afraid           | we 're not afraid no we are not afraid we 'll not be |

#### Sample of `compound` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                 |
|:-----------|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------------|
| 63-65-66   | nobody           | so               | lonely           | previously unreleased ) / NOBODY / SO LONELY ( alternate version unreleased |
| 41-42-43   | none             | too              | benevolent       | the very least , none too benevolent ) course .                             |
| 56-57-58   | none             | too              | subtle           | a sweet ( but none too subtle ) metaphor for the                            |

#### Sample of `expl` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                    |
|:-----------|:-----------------|:-----------------|:-----------------|:-----------------------------------------------|
| 3-5-6      | nit              | equally          | silly            | But is nit it equally silly that in Victoria , |
| 2-4-5      | nit              | more             | exciting         | Is nit it more exciting to play with your      |

#### Sample of `partmod` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                      |
|:-----------|:-----------------|:-----------------|:-----------------|:-------------------------------------------------------------------------------------------------|
| 38-46-47   | nothing          | very             | trustworthy      | sense that , being nothing more than elaborate socks , they are very trustworthy , and even more |


In [11]:
print(df_clean[df_clean.neg_deprel=='preconj'].sample(20).filter(regex=r'(lower|window)').reset_index(drop=True).to_markdown())

|    | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                            |
|---:|:-----------------|:-----------------|:-----------------|:---------------------------------------------------------------------------------------|
|  0 | neither          | particularly     | unique           | ' Clintonesque statements were neither particularly unique nor compelling , the        |
|  1 | neither          | really           | bad              | kiddie animation that 's neither really bad nor good .                                 |
|  2 | neither          | exceptionally    | large            | telescope whose mirror is neither exceptionally large -LRB- albeit slightly bigger     |
|  3 | neither          | particularly     | difficult        | adding that recounting was neither particularly difficult nor costly .                 |
|  4 | neither          | too              | hot              | '' economy that is

|    | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                       |
|---:|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------------------|
|  0 | neither          | too              | hot              | an economy that proved neither too hot nor too cold .                             |
|  1 | neither          | even             | informal         | as normal English , neither slang nor even informal .                             |
|  2 | neither          | as               | cheap            | have suffered by being neither as cheap as the discounters nor                    |
|  3 | neither          | as               | targeted         | tax cut that is neither as targeted as the Democrats would                        |
|  4 | neither          | suitably         | christian        | implication that he is neither suitably Christian nor American in his             |
|  5 | neither          | excruciatingly   | thick            | Eddie Kean , was neither excruciatingly thick like Mortimer nor wickedly          |
|  6 | neither          | too              | tired            | but McGrath is neither too tired nor too shocked to                               |
|  7 | neither          | particularly     | tender           | braised lamb shank is neither particularly tender nor particularly flavorful ,    |
|  8 | neither          | too              | fast             | a rate that 's neither too fast nor too slow .                                    |
|  9 | neither          | overtly          | supportive       | released a prepared statement neither overtly supportive or critical .            |
| 10 | neither          | as               | severe           | the coming weeks is neither as severe as those in many                            |
| 11 | neither          | too              | hard             | Goldilocks charm of being neither too hard nor too soft ,                         |
| 12 | neither          | too              | academic         | the insightful analyses are neither too academic nor pedantic and are             |
| 13 | neither          | officially       | catholic         | the 300-member parish was neither officially Catholic nor any longer Episcopalian |
| 14 | neither          | as               | common           | general practitioners are neither as common nor as general as                     |
| 15 | neither          | too              | hot              | where conditions should be neither too hot nor too cold but                       |
| 16 | neither          | too              | sumptuous        | lived-in rooms that looked neither too sumptuous nor too new ,                    |
| 17 | neither          | as               | good             | available at supermarkets was neither as good as the fries sold                   |
| 18 | neither          | too              | regular          | prefer patterns that are neither too regular , like the test                      |
| 19 | neither          | as               | harmful          | a drug that is neither as harmful nor as addictive as                             |
